Arm/Decode.lean

```diff
@@ -104,6 +106,11 @@ def decode_data_proc_sfp (i : BitVec 32) : Option ArmInst :=
     DPSFP (Advanced_simd_permute {Q, size, Rm, opcode, Rn, Rd})
   | [0, Q:1, op:1, 0111100000, a:1, b:1, c:1, cmode:4, o2:1, 1, d:1, e:1, f:1, g:1, h:1, Rd:5] =>
     DPSFP (Advanced_simd_modified_immediate {Q, op, a, b, c, cmode, o2, d, e, f, g, h, Rd})
+  -- Note: Advanced SIMD shift by immediate constraint immh != 0000
```

Comment:
Where is this constraint enforced?

Reply:
I added the note because it is not enforced but inferred. Currently an instruction will be matched against `Advanced_simd_modified_immediate` first, if it doesn't match, then we know `immh` can't be `0b0000#4`.

Reply:
We will need a more capable `match_bv` to be able to supply `immh != 0b0000#4` as a guard.

Reply:
Ah, I understand! We can, of course, make `match_bv` take a guard if we come across more of these kinds of issues (and care about even more readability for this decoding), but for now, it serves the purpose well enough.

I'd recommend one of two things to resolve this: 
1. Modify the comment for clarity. i.e., essentially paste in what you say in your first comment above.
2. Delete the comment altogether (I found its existence a little confusing).